src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,44 @@
+package mate.academy.service.impl;
+
+import java.util.Optional;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.AuthenticationService;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> userFromDbOptional = userService.findByEmail(email);
+        if (userFromDbOptional.isPresent()) {
+            User userFromDB = userFromDbOptional.get();
+            String passwordFromDb = userFromDB.getPassword();
+            String hashPassword = HashUtil.hashPassword(password, userFromDB.getSalt());
+            boolean isValidPassword = !password.isEmpty() && hashPassword.equals(passwordFromDb);
```

Comment:
why do you need `!password.isEmpty()`?
use `password != null`
isEmpty() can be used in `register` but not necessarily.